api/swapi.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+
+const got = require('got');
+
+const paths = {
+    people: 'people/',
+    films: 'films/',
+    species: 'species/',
+    planet: 'planets/'
+}
+
+
+class Swapi {
+
+    constructor(path) {
+        this.baseUrl = 'https://swapi.co/api/';
+        this.objectId = path;
+        this.gotDefaultOption = {
+            json: true
+        };
+    }
+
+    async get(url, options = this.gotDefaultOption) {
+        return got(url, options);
+    }
+
+    async getDetailedCharacter(characterId) {
+        const url = `${this.baseUrl}${paths.people}${characterId}`;
+        
+        try {
+            const character = await this.get(url);
+            const species = await this.get(character.body.species[0]);
+            const homeWorld = await this.get(character.body.homeworld);
+            const films = await Promise.all(character.body.films.map(film=>this.get(film)));
+            return { ...character.body, homeworld: homeWorld.body, species: species.body, films: films.map((film)=>film.body) };
+        } catch (e) {
+            console.log(e)
+        }
+    }
+
+
+    // async getCharacter({ id = 0, options = gotDefaultOption, url = null }) {
```

Comment:
A few things:

1) I don't like passing all these params as an object. They should be individual params
2) The `url` option should not be on this. If the dev already has the exact url, they can just call `Swapi.get(url)` instead.
3) There should not be a default for `id`. Whoever is calling the function should be forced to provide an id. Defaulting to the API resource with id 0 is not expected behaviour